python/foglamp/common/configuration_manager.py

```diff
@@ -764,37 +764,17 @@ def _str_to_ipaddress(item_val):
             except ValueError:
                 return False
 
-        def _str_to_json(item_val):
-            if _str_to_bool(item_val) or _str_to_int(item_val) or _str_to_ipaddress(
-                    item_val) or _str_to_password(item_val) or _str_to_x509cert(item_val):
-                return False
-            try:
-                json.loads(item_val)
-            except Exception:
-                return False
-            else:
-                return True
-
-        def _str_to_password(v):
-            # TODO:
-            pass
-
-        def _str_to_x509cert(v):
-            # TODO:
-            pass
-
         if _type == 'boolean':
             return _str_to_bool(_value)
         elif _type == 'integer':
             return _str_to_int(_value)
         elif _type == 'JSON':
-            return _str_to_json(_value)
+            return Utils.is_json(_value)
         elif _type == 'IPv4' or _type == 'IPv6':
             return _str_to_ipaddress(_value)
 
     def _clean(self, item_type, item_val):
         if item_type == 'boolean':
             return item_val.lower()
-        if item_type == 'JSON':
-            return json.dumps(json.loads(item_val))
```

Comment:
to solve the 404: No detail found for the category_name: x and config_item: y
```
if item_type == 'JSON':
    return json.loads(item_val)

```

Reply:
For test__validate_category_val_valid_type

```       
 if test_input == "JSON":
            assert c_return_value["test_item_name"]["value"] == json.loads(test_value)
        else:
            assert c_return_value["test_item_name"]["value"] == test_value
```

I'm not sure if we should start returning and accepting the values for boolean, integer value in type based format itself. Else you shall need to use json.dumps for JSON before returning

Reply:
https://github.com/foglamp/foglamp-gui/pull/210/files to have quick test and curl you provided!